templates/terraform/examples/pubsub_subscription_dead_letter.tf.erb

```diff
@@ -0,0 +1,17 @@
+resource "google_pubsub_topic" "<%= ctx[:primary_resource_id] %>" {
+  name = "<%= ctx[:vars]['topic_name'] %>"
+}
+
+resource "google_pubsub_topic" "<%= ctx[:primary_resource_id] %>_dead_letter" {
+  name = "<%= ctx[:vars]['topic_name'] %>"
+}
+
+resource "google_pubsub_subscription" "<%= ctx[:primary_resource_id] %>" {
+  name  = "<%= ctx[:vars]['subscription_name'] %>"
+  topic = google_pubsub_topic.<%= ctx[:primary_resource_id] %>.name
+
+  expiration_policy {
+    dead_letter_topic = google_pubsub_topic.<%= ctx[:primary_resource_id] %>_dead_letter.id
+    max_delivery_attempts = <%= ctx[:vars]['max_delivery_attempts'] %>
```

Comment:
The issue is that this is getting a random suffix added on during generation. Instead of using ctx here, just put the number 10 directly in this config

Reply:
Fixed